rust/lance-encoding/src/encodings/physical/bitmap.rs

```diff
@@ -92,24 +93,20 @@ struct BitmapDecoder {
     chunks: Vec<BitmapData>,
 }
 
-impl PhysicalPageDecoder for BitmapDecoder {
-    fn update_capacity(
-        &self,
-        _rows_to_skip: u32,
-        num_rows: u32,
-        buffers: &mut [(u64, bool)],
-        _all_null: &mut bool,
-    ) {
-        buffers[0].0 = arrow_buffer::bit_util::ceil(num_rows as usize, 8) as u64;
-        buffers[0].1 = true;
-    }
-
+impl PrimitivePageDecoder for BitmapDecoder {
     fn decode_into(
         &self,
         rows_to_skip: u32,
         num_rows: u32,
-        dest_buffers: &mut [BytesMut],
-    ) -> Result<()> {
+        _all_null: &mut bool,
+    ) -> Result<Vec<BytesMut>> {
+        let mut capacities = vec![(0, false); self.num_buffers() as usize];
+
+        capacities[0].0 = arrow_buffer::bit_util::ceil(num_rows as usize, 8) as u64;
+        capacities[0].1 = true;
+
+        let mut dest_buffers = create_buffers_from_capacities(capacities);
+
         let mut rows_to_skip = rows_to_skip;
 
         let mut dest_builder = BooleanBufferBuilder::new(num_rows as usize);
```

Comment:
Hmm...going forward I think we can actually make this zero-copy now.  But lets save that for a future PR :)